cupy/core/fusion.py

```diff
@@ -496,84 +625,72 @@ def _get_fix_code(data_type, fixed_type, operation):
     return module_code
 
 
-def _get_fusion(func, nin, reduce, post_map, identity, input_types, name):
-    in_vars = [_FusionVar(i, t) for i, t in enumerate(input_types)]
-    mem = _FusionMem(in_vars)
-    in_refs = [_FusionRef(_, mem) for _ in in_vars]
-    out_refs = func(*in_refs)
-    out_refs = list(out_refs) if type(out_refs) == tuple else [out_refs]
-    out_refs = [_ for _ in out_refs if _ is not None]
-    out_refs = [_FusionRef(_normalize_arg(_, mem), mem) for _ in out_refs]
-    out_vars = [_normalize_arg(copy(_), mem) for _ in out_refs]
-    nout = len(out_vars)
-    op_list = mem.op_list
-    tmpvars = mem.var_list[len(in_vars):]
-    if nout > 0:
-        tmpvars = tmpvars[:-nout]
-
-    in_params = ', '.join(_get_params(in_vars[:nin]))
-    out_params = ', '.join(_get_params(out_vars))
-    operation = ''.join(_get_declaration_from_var(_) for _ in tmpvars)
-    operation += ''.join(_get_declaration_from_op(_) for _ in op_list)
-    operation += '\n'.join(_get_operation_code(_) for _ in op_list)
-
-    if reduce is None:
-        if not out_params:
-            in_params = ', '.join(_get_params(in_vars[:-1]))
-            out_params = ', '.join(_get_params([in_vars[-1]]))
-        submodules = _gather_submodules(op_list)
-        submodule_code = ''.join(_get_submodule_code(_)
-                                 for _ in submodules.values())
-        return core.ElementwiseKernel(in_params, out_params,
-                                      operation, preamble=submodule_code,
-                                      name=name)
+def _get_fusion_from_types(func, in_types, name):
```

Comment:
Write a few lines of comments explaining what this function does.
Especially, what's the meaning of the return values?

Reply:
Documented.